xapian-letor/tests/api_letor.cc

```diff
@@ -833,3 +914,456 @@ DEFINE_TESTCASE(ndcg_score_test, generated)
     unlink("ndcg_score_test.txt");
     return true;
 }
+
+// Test createfeaturevector method for TfFeature
+DEFINE_TESTCASE(createfeaturevector_tffeature, generated)
+{
+    vector<Xapian::Feature*> f;
+    Xapian::TfFeature* f1 = new Xapian::TfFeature();
+    f.push_back(f1);
+
+    Xapian::FeatureList fl(f);
+    Xapian::Database db = get_database("db_index_three_documents",
+				       db_index_three_documents);
+    Xapian::Enquire enquire(db);
+    enquire.set_query(Xapian::Query("score"));
+    Xapian::MSet mset;
+    mset = enquire.get_mset(0, 10);
+
+    TEST(!mset.empty());
+
+    Xapian::QueryParser queryparser;
+    queryparser.set_stemmer(Xapian::Stem("en"));
+    queryparser.set_stemming_strategy(queryparser.STEM_ALL_Z);
+    queryparser.add_prefix("title", "S");
+    queryparser.add_prefix("description", "XD");
+
+    string querystring = "title:score description:score score";
+    Xapian::Query query = queryparser.parse_query(querystring);
+
+    auto fv = fl.create_feature_vectors(mset, query, db);
+    TEST_EQUAL(fv.size(), 2);
+    TEST_EQUAL(fv[0].get_fcount(), 4);
+    TEST_EQUAL(fv[1].get_fcount(), 4);
+
+    vector<double> fvals_doc1 = fv[0].get_fvals();
+    vector<double> fvals_doc2 = fv[1].get_fvals();
+    TEST_EQUAL(fvals_doc1.size(), 4);
+    TEST_EQUAL(fvals_doc2.size(), 4);
+
+    vector<double> test_vals_doc1(4);
+    test_vals_doc1[0] = 0.301029995663981;
+    test_vals_doc1[1] = 1.653212513775344;
+    test_vals_doc1[2] = 1.954242509439325;
```

Comment:
I think we need a brief comment explaining where these numbers come from. (Even though the answer is just that they are the appropriate TF values for the first document.)

Reply:
fixed in a fixup commit.